devices/MAX1704x/Max1704X.cs

```diff
@@ -0,0 +1,307 @@
+        /// <returns>The voltage of the battery in volts.</returns>
+        public float BatteryVoltage
+        {
+            get
+            {
+                var vCell = Read16(Registers.Max17043Vcell);
+                return vCell / Divider * FullScale;    
+            }
+        }
+
+        /// <summary>
+        /// Gets the battery percentage.
+        /// </summary>
+        /// <returns>The battery percentage as a float value.</returns>
+        public float BatteryPercent
+        {
+            get
+            {
+                var soc = Read16(Registers.Max17043Soc);
+                var percent = (float)((soc & 0xFF00) >> 8);
+                percent += (soc & 0x00FF) / 256.0f;
+                return percent;    
+            }
+        }
+
+        /// <summary>
+        /// Gets the version of the device.
+        /// </summary>
+        /// <value>The version of the device.</value>
+        public ushort Version => Read16(Registers.Max17043Version);
+
+        /// <summary>
+        /// Gets or sets the threshold of the property.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the value is out of range.</exception>
+        public byte Threshold
+        {
+            get
+            {
+                var configReg = GetConfigRegister();
+                var threshold = configReg & 0x001F;
+                threshold = 32 - threshold;
+                return (byte)threshold;    
+            }
+
+            set
+            {
+                if (value > 100 || value < ThresholdMinValue)
+                {
+                    throw new ArgumentOutOfRangeException();
+                }
+                
+                var percent = ThresholdMinValue - value;
+                var configReg = GetConfigRegister();
+                configReg &= 0xFFE0;
+                configReg |= (ushort)percent;
+                Write16(configReg, Registers.Max17043Config);
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether an alert is active.
+        /// </summary>
+        /// <value>
+        /// <c>true</c> if an alert is active; otherwise, <c>false</c>.
+        /// </value>
+        public bool Alert
+        {
+            get
+            {
+                var configReg = GetConfigRegister();
+                if ((configReg & Registers.Max17043ConfigAlert) != 0)
+                {
+                    return true;
+                }
+
+                return false;    
+            }
+        }
+
+        /// <summary>
+        /// Gets the maximum value that the property can have or represent.
+        /// </summary>
+        protected abstract float FullScale { get; }
+
+        /// <summary>
+        /// Gets the value used to divide two numbers in the implementation of a division operation.
+        /// </summary>
+        protected abstract float Divider { get; }
+
+        private readonly I2cDevice _i2CDevice;
```

Comment:
can you move this one at the beginning so we have all the variables all together?

Reply:
done